src/Components/Components/src/Auth/AuthorizeView.razor

```diff
@@ -0,0 +1,68 @@
+@namespace Microsoft.AspNetCore.Components
+
+@if (currentAuthState == null)
+{
+    @Authorizing
+}
+else if (IsAuthorized())
+{
+    @((Authorized ?? ChildContent)?.Invoke(currentAuthState))
+}
+else
+{
+    @NotAuthorized
+}
+
+@functions {
+    private IAuthenticationState currentAuthState;
+
+    [CascadingParameter] public Task<IAuthenticationState> AuthStateTask { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed if the user is authorized.
+    /// </summary>
+    [Parameter] public RenderFragment<IAuthenticationState> ChildContent { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed if the user is not authorized.
+    /// </summary>
+    [Parameter] public RenderFragment NotAuthorized { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed if the user is authorized.
+    /// If you specify a value for this parameter, do not also specify a value for <see cref="ChildContent"/>.
+    /// </summary>
+    [Parameter] public RenderFragment<IAuthenticationState> Authorized { get; private set; }
+
+    /// <summary>
+    /// The content that will be displayed while asynchronous authorization is in progress.
+    /// </summary>
+    [Parameter] public RenderFragment Authorizing { get; private set; }
+
+    protected override async Task OnParametersSetAsync()
```

Comment:
This will eventually have to accept an auth policy name or auth policy. I assume you're waiting for that relayering to blow over first :)

Reply:
Yes, exactly. This is also mentioned in the `IsAuthorized` logic below.